### PR TITLE
HDDS-9713. Replace Thread.sleep in TestPipelineClose.

### PR DESCRIPTION
**Pull request by nandakumar131**
## What changes were proposed in this pull request?
Replace Thread.sleep in TestPipelineClose#testPipelineCloseWithPipelineAction with a conditional wait.

## What is the link to the Apache JIRA
HDDS-9713

## How was this patch tested?
This is a unit test change

### CODE REVIEW

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestPipelineClose.java [hunk 1]
```diff
@@ -165,35 +166,41 @@ public void testPipelineCloseWithOpenContainer()
...
+        .waitFor(() -> {
+          final List<PipelineReport> pipelineReports = ozoneContainer
+              .getPipelineReport().getPipelineReportList();
+          for (PipelineReport pipelineReport : pipelineReports) {
+            if (pipelineReport.getPipelineID().equals(pid)) {
+              return false;
+            }
+          }
+          return true;
+        }, 500, 5000);
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestPipelineClose.java:198]: nit: unnecessary indent

```suggestion
    GenericTestUtils.waitFor(() -> {
      final List<PipelineReport> pipelineReports = ozoneContainer
          .getPipelineReport().getPipelineReportList();
      for (PipelineReport pipelineReport : pipelineReports) {
        if (pipelineReport.getPipelineID().equals(pid)) {
          return false;
        }
      }
      return true;
    }, 500, 5000);
```

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestPipelineClose.java [hunk 2]
```diff
@@ -165,35 +166,41 @@ public void testPipelineCloseWithOpenContainer()
...
 
     try {
       pipelineManager.getPipeline(ratisContainer.getPipeline().getId());
-      Assert.fail("Pipeline should not exist in SCM");
-    } catch (PipelineNotFoundException e) {
+      Assertions.fail("Pipeline should not exist in SCM");
+    } catch (PipelineNotFoundException ignored) {
     }
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestPipelineClose.java:204]: We should change all assertions to JUnit5 in the file, but I'd prefer to do it in a separate task (as part of umbrella: HDDS-6729).  Having both `Assert` and `Assertions` will be confusing.

Also, `try-catch` can be replaced with `assertThrows()`.

**Review by adoroszlai — COMMENTED** [review]
Thanks @nandakumar131 for the improvement.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestPipelineClose.java (hunk 2 above)
Comment (nandakumar131) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestPipelineClose.java:204]: I agree, we can do the JUnit5 changes in a separate jira.

**Comment by nandakumar131** [comment]
@adoroszlai, thanks for taking a look at the changes. I have addressed the review comments.

**Review by adoroszlai — APPROVED** [review]
Thanks @nandakumar131 for updating the patch.

**Pull request merged**